core/src/cost_model.rs

```diff
@@ -0,0 +1,136 @@
+//! `cost_model` aiming to limit the size of broadcasting sets, and reducing the number
+//! of un-parallelizeble transactions (eg, transactions as same writable key sets).
+//! By doing so to improve leader performance.
+
+use solana_sdk::{
+    bpf_loader, bpf_loader_deprecated, bpf_loader_upgradeable, pubkey::Pubkey, system_program,
+};
+use solana_runtime::{
+    hashed_transaction::HashedTransaction,
+};
+use std::{collections::HashMap, str::FromStr};
+
+// TODO  revisit these hardcoded numbers, better get from mainnet log
```

Comment:
below are just some made-up numbers, will be replace with stats collected from mainnet later